### PR TITLE
chore: integrate react-native-community/eslint standards

### PR DESCRIPTION
**Pull request by mikehardy**
## Description

react-native-community has some nascent standards, and [the use of lint is one of them](https://github.com/react-native-community/.github/issues/9).

I've patterned this off the [work in react-native-maps](https://github.com/react-native-community/react-native-maps/pull/2862), netinfo etc - [guidance here](https://github.com/facebook/react-native/tree/master/packages/eslint-config-react-native-community)

Basically, this installs the requisite lint package and since we had no lint standard before, I just extended the community standard in in package.json

Our prettierrc was already similar, so there were very only 4 errors to fix

To ensure we don't skew over time, I installed lint-staged and husky, and configured the module to run eslint in fix mode as a commit hook

### CODE REVIEW

**Comment by mikehardy** [comment]
This begins the implementation of #307

**Comment by mikehardy** [comment]
@machour hope you're enjoying the respite now that you're less solo in here :-) - since I just changed a pack of dependencies and how the commit flow works even, I'd love if you pulled this branch and tried it (test commit something or similar). Thanks!

**Comment by mikehardy** [comment]
Cool - I'll rebase and merge it - thanks!
